15-may-17/solutions/simon/lib/look_and_say.ex

```diff
@@ -0,0 +1,26 @@
+defmodule LookAndSay do
+  @moduledoc """
+  Documentation for LookAndSay.
+  """
+
+  @spec generate_look_and_say_sequence(str :: String.t) :: String.t
+  def generate_look_and_say_sequence(str) do
+    Regex.scan(~r/(\d)\1+|(\d)/, str)
+    |> Enum.map(&regex_to_say_representation/1)
+    |> Enum.join
+  end
+
+  @spec regex_to_say_representation(list :: list) :: String.t
+  defp regex_to_say_representation(list) do
+    list
+    |> Enum.reject(fn(x) -> x == "" end)
```

Comment:
So basically, the regex returns `["1", "", "1"]` if a single digit is matched. I wonder if the regex could be updated to just return two regardless of whether one or more are matched.

I mean, for me I personally like the use of the regex for this solution. It does all the hard lifting. Arguably, you are right, it is on the cusp of whether regex is right or not.